src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1280,32 +1262,21 @@ public long ToFileTimeUtc()
 
         public DateTime ToLocalTime()
         {
-            return ToLocalTime(false);
-        }
-
-        internal DateTime ToLocalTime(bool throwOnOverflow)
-        {
-            if (Kind == DateTimeKind.Local)
+            if ((_dateData & LocalMask) != 0)
```

Comment:
>LocalMask [](start = 29, length = 9)

please add comment we are testing for local kind or convert this check to inlined property.